modules/nw-cudn-localnet.adoc

```diff
@@ -0,0 +1,79 @@
+= Creating a ClusterUserDefinedNetwork CR for Localnet
+
+A `Localnet` topology connects the secondary network to the physical underlay. This enables both east-west cluster traffic and access to services running outside the cluster, but it requires additional configuration of the underlying Open vSwitch (OVS) system on cluster nodes.
+
+.Prerequisites
+
+* You have logged in as a user with `cluster-admin` privileges.
+
+* You have created and configured the Open vSwitch (OVS) bridge mapping to associate the logical OVN-Kubernetes network with the physical node network through the OVS bridge. For more information, see "Configuration for a localnet switched topology".
+
+.Procedure
+
+. Create a request for `Localnet` topology type cluster-wide user-defined network:
+
+.. Create a YAML file, such as `cluster-udn-localnet.yaml`, to define your request for a `localnet` topology as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: <cudn_name> # <1>
+spec:
+  namespaceSelector: # <2>
+    matchLabels: # <3>
+      "<label_1_key>": "<label_1_value>" # <4>
+      "<label_2_key>": "<label_2_value>" # <4>
+  network: # <5>
+    topology: Localnet # <6>
+    localnet: # <7>
+      role: Secondary # <8>
+      physicalNetworkName: test
+      ipam: {lifecycle: Persistent}
+      subnets: ["192.168.0.0/16", "2001:dbb::/64"] # <9>
+----
+<1> Name of your `ClusterUserDefinedNetwork` CR.
+<2> A label query over the set of namespaces that the cluster UDN CR applies to. Uses the standard Kubernetes `MatchLabel` selector. Must not point to `default` or `openshift-*` namespaces.
+<3> Uses the `matchLabels` selector type, where terms are evaluated with an `AND` relationship.
+<4> Because the `matchLabels` selector type is used, provisions namespaces that contain both `<label_1_key>=<label_1_value>` and `<label_2_key>=<label_2_value>` labels.
+<5> Describes the network configuration.
+<6> The `topology` field describes the network configuration. Specifying a `Localnet` topology type creates one logical switch that is shared by all nodes.
+<7> This field specifies the topology configuration. It must be `localnet`.
+<8> Specifies the `role` for the network configuration. `Secondary` is the only `role` specification supported for the `localnet` topology.
+<9> For `Localnet` topology types the following specifies config details for the `subnet` field:
++
+* The subnets field is optional.
+* The subnets field is of type `string` and accepts standard CIDR formats for both IPv4 and IPv6.
+* The subnets field accepts one or two items. For two items, they must be of a different family. For example, subnets values of `10.100.0.0/16` and `2001:db8::/64`.
+* `Localnet` subnets can be omitted. If omitted, users must configure static IP addresses for the pods. As a consequence, port security only prevents MAC spoofing. For more information, see "Configuring pods with a static IP address".
```

Comment:
Where can I see the example for static IP address?

Reply:
Hmm, I'd also say that it should be acceptable to have an external DHCP server, such that we don't specify the subnets but we can still have the pods use DHCP and they get the IPs from the external DHCP server. However, not sure if somebody from eng can confirm that my understanding is correct.

Reply:
Yes, @palonsoro is correct, you can have a DHCP server on your network (either in the physical network, or just a pod implementing the DHCP server in the logical network) to advertise IPs for the network's workloads. 

However, this also adds requirements to the workloads: it needs to listen to the DHCP offers. I don't think a "regular" pod does this.

Reply:
These are all good points. I _think_ the place to provide the examples and further details that you all are discussing though is in the virtualization docs not the core-networking docs. I'll see what the cnv and virt team have planned for `localnet` in 4.19.

Reply:
I'm borderline **sure** that's described in the virt docs. 

Still, let's ask @sjhala-ccs :)

Reply:
@maiqueb @JoeAldinger I will start work on the localnet story ([CNV-52855](https://issues.redhat.com/browse/CNV-52855)) soon. Here's what we currently have in the virt docs on OVN-K localnet: https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/virtualization/networking#virt-creating-localnet-nad-cli_virt-connecting-vm-to-ovn-secondary-network

Reply:
So there's this [well placed note](https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/virtualization/networking#virt-creating-localnet-nad-cli_virt-connecting-vm-to-ovn-secondary-network) in the virt docs for **layer2** topologies:
> The above example configures a cluster-wide overlay without a subnet defined. This means that the logical switch implementing the network only provides layer 2 communication. You must configure an IP address when you create the virtual machine by either setting a static IP address or by deploying a DHCP server on the network for a dynamic IP address. 

I think we should carry over that note to the localnet docs - it also applies there.

Reply:
Makes sens to me

Reply:
@maiqueb you mean in the virt docs, right? Given that the note itself references OCP virt, a use case that would be for virt/vms, and the KCS is a virt use case as well. If so, I'm going to resolve this conversation and leave my colleague in virt to pick the note up in their `localnet` CUDN docs @sjhala-ccs